From d1210dbb39e725f82f298a0f3442bcc61b63f09f Mon Sep 17 00:00:00 2001
From: Fotios Sioutis <sfotis@gmail.com>
Date: Tue, 22 Mar 2011 07:18:25 +0100
Subject: [PATCH 8/9] Move code to work around a BCC bug

For an unknown reason, BCC reports this error:
  [C++ Error] Standard_ErrorHandler.cpp(48): E2092 Storage class 'static' is not allowed here
which disappears when GetThreadID() is defined before static members.
---
 ros/src/Standard/Standard_ErrorHandler.cxx |   17 +++++++++--------
 1 files changed, 9 insertions(+), 8 deletions(-)

From 63bfacb18355d6e03bc3098c50f25486e533acb1 Mon Sep 17 00:00:00 2001
From: Fotios Sioutis <sfotis@gmail.com>
Date: Fri, 18 Nov 2011 12:03:23 +0200
Subject: [PATCH] Fixed convertion from int to Handle under bcc32

---
 inc/AIS_InteractiveContext.lxx |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

From 29c06652969463d012eb97990f07a8bf6feeae77 Mon Sep 17 00:00:00 2001
From: Fotios Sioutis <sfotis@gmail.com>
Date: Fri, 11 Nov 2011 23:34:40 +0200
Subject: [PATCH] Resolves a possible bcc32 compiler bug.

For a strange reason this file must not have
exception handling compiler option enabled,
otherwise an internal compiler error occurs.
---
 src/ShapeFix/ShapeFix_ComposeShell.cxx |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

Index: oce/src/AIS/AIS_InteractiveContext.lxx
===================================================================
--- oce.orig/src/AIS/AIS_InteractiveContext.lxx
+++ oce/src/AIS/AIS_InteractiveContext.lxx
@@ -96,7 +96,7 @@ inline Standard_Boolean AIS_InteractiveC
 {return myCurLocalIndex != 0;}
 
 inline Handle_AIS_LocalContext AIS_InteractiveContext::LocalContext() const
-{ return (myCurLocalIndex > 0) ? myLocalContexts(myCurLocalIndex) : NULL; }
+{ return (myCurLocalIndex > 0) ? myLocalContexts(myCurLocalIndex) : (Handle_AIS_LocalContext) NULL; }
 
 inline const TCollection_AsciiString& AIS_InteractiveContext::CurrentName() const 
 {return myCurrentName;} 
Index: oce/src/ShapeFix/ShapeFix_ComposeShell.cxx
===================================================================
--- oce.orig/src/ShapeFix/ShapeFix_ComposeShell.cxx
+++ oce/src/ShapeFix/ShapeFix_ComposeShell.cxx
@@ -22,6 +22,10 @@
 //    abv  22.07.99 implementing patch indices
 //    svv  10.01.00 porting on DEC
 
+#ifdef __BORLANDC__
+#pragma option -x-
+#endif
+
 #include <ShapeFix_ComposeShell.ixx>
 
 #include <Precision.hxx>
Index: oce/src/Standard/Standard_ErrorHandler.cxx
===================================================================
--- oce.orig/src/Standard/Standard_ErrorHandler.cxx
+++ oce/src/Standard/Standard_ErrorHandler.cxx
@@ -41,14 +41,7 @@
 // request will be ignored. If the abort request do a raise during a setjmp
 // or a longjmp, there will be a "terminating SEGV" impossible to handle.
 
-//==== The top of the Errors Stack ===========================================
-static Standard_ErrorHandler* Top = 0;
-
-// A mutex to protect from concurrent access to Top
-// Note that we should NOT use Sentry while in this class, as Sentry
-// would register mutex as callback in the current exception handler
-static Standard_Mutex theMutex; 
-
+//Somehow borland needs this inline global function to be declared first ... ??
 static inline Standard_ThreadId GetThreadID()
 {
 #ifndef WNT
@@ -58,6 +51,14 @@ static inline Standard_ThreadId GetThrea
 #endif
 }
 
+//==== The top of the Errors Stack ===========================================
+static Standard_ErrorHandler* Top = 0;
+
+// A mutex to protect from concurrent access to Top
+// Note that we should NOT use Sentry while in this class, as Sentry
+// would register mutex as callback in the current exception handler
+static Standard_Mutex theMutex; 
+
 //============================================================================
 //====  Constructor : Create a ErrorHandler structure. And add it at the 
 //====                'Top' of "ErrorHandler's stack".
